RELEASE.md

```diff
@@ -12,6 +12,7 @@
 * Upgraded `pip-tools`, which is used by `kedro build-reqs`, to 6.4. This `pip-tools` version requires `pip>=21.2` while [adding support for `pip>=21.3`](https://github.com/jazzband/pip-tools/pull/1501). To upgrade `pip`, please refer to [their documentation](https://pip.pypa.io/en/stable/installing/#upgrading-pip).
 * Extended ``ExcelDataSet`` to support saving Excel files with multiple sheets.
 * `kedro pipeline package <pipeline>` now raises an error if the `<pipeline>` argument doesn't look like a valid Python module path (e.g. has `/` instead of `.`).
+* Introduced `pandas.GenericDataSet` which provides a 'best effort' facility to read / write any format provided by the `pandas` library without a specific Kedro dataset existing beforehand. For a given `kind` parameter, this dataset requires that at least one `pd.read_{kind}` or `pd.DataFrame.to_{kind}` method is available for use.
```

Comment:
Please can we move this to the table in "Major features and improvements" to go with the other new dataset in this release?